tests/slepos/build_images_creator.pm

```diff
@@ -0,0 +1,254 @@
+    while (!check_screen('image-created', $checktime) && !$timeout) {
+        $buildtime += $checktime;
+        $timeout = 1 if $buildtime > $maxbuildtime;
+        mouse_set(($buildtime / 2) % 800, 0);
+    }
+
+    if ($timeout) {            #timeout - save log
+        die "Error: image creation timeout (${maxbuildtime}s), image build failed!";
+    }
+
+    #save logs
+    send_key('alt-s');
+    assert_screen('save-image-creation-logs');
+    send_key('alt-t');
+    send_key('home');
+    send_key('shift-end');
+    type_string($image . '.stdout');
+    send_key('alt-e');
+    send_key('home');
+    send_key('shift-end');
+    type_string($image . '.stderr');
+    save_screenshot;
+    send_key('alt-s');
+
+    assert_screen('image-created');
+    send_key('alt-o');
+    assert_screen('image-created-ok');
+    assert_and_click('ok-checkbox');
+    assert_screen('image-creator-addimage');    #back to start ascreen
+
+    #upload image
+    select_console('root-console');
+    wait_still_screen;
+    save_screenshot;
+    type_string "mv /var/lib/SLEPOS/system/images/$imagebase/ /var/lib/SLEPOS/system/images/$image/\n";
+    type_string "rm -fr /var/lib/SLEPOS/system/$imagebase\n";    #remove previous cfg to allow subsequent images of same basename
+    script_output "ls -l /var/lib/SLEPOS/system/images/$image/";
+    script_output "ls -l /tmp/$image*";
+    upload_logs "/tmp/$image.stderr",                                                  'public';
+    upload_logs "/tmp/$image.stdout",                                                  'public';
+    script_output "tar -cjf $image.$img_suffix /var/lib/SLEPOS/system/images/$image/", 2000;
+    upload_asset "$image.$img_suffix",                                                 'public';
+
+    select_console('x11');
+    ensure_unlocked_desktop;
+
+}
+
+
+sub prepare_template {
+
+    my ($template) = @_;
+
+    my $image_path = '/usr/share/kiwi/image/SLEPOS';
+    my @vdrivers   = (
+        'drivers/virtio/virtio_ring.ko', 'drivers/virtio/virtio.ko', 'drivers/net/virtio_net.ko', 'drivers/scsi/virtio_scsi.ko',
+        'drivers/block/virtio_blk.ko',   'drivers/virtio/virtio_pci.ko'
+    );
+
+    type_string
+      "grep -v \"wireless support\" \"$image_path/$template/config.xml\" |grep -v \"SUSE Manager support\" > \"$image_path/$template/config.xml.tmp\"\n";
+    type_string "mv -f \"$image_path/$template/config.xml.tmp\" \"$image_path/$template/config.xml\"\n";
+
+    my $driverlist;
+    foreach my $driver (@vdrivers) {
+        $driverlist .= "<file name='$driver'/>";
+    }
+    script_output "sed -i -e \"s|</drivers>|$driverlist</drivers>|\" \"$image_path/$template/config.xml\"";
+    type_string "cat $image_path/$template/config.xml\n";
+    wait_still_screen;
+    save_screenshot;
+
+}
+
+
+sub run() {
+
+    #todo: generalize so more than just graphical can be built
+    my $images_ref = get_var_array('IMAGE_CREATOR');
+
+    save_screenshot;
+    select_console('x11');
```

Comment:
that will now by default assert the console '11' to show up, e.g. "generic-desktop". Because you are using "ensure_unlocked_desktop" afterwards I recommend to say `select_console 'x11', await_console => 0;`